packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -131,6 +149,76 @@ function ListView<T extends object>(props: ListViewProps<T>, ref: DOMRef<HTMLDiv
+      renderPreview(selectedKeys, draggedKey) {
+        let item = state.collection.getItem(draggedKey);
+        let itemWidth = domRef.current.offsetWidth;
+        let showCheckbox = state.selectionManager.selectionMode !== 'none' && state.selectionManager.selectionBehavior === 'toggle';
+        let isSelected = state.selectionManager.isSelected(item.key);
+        return (
+          <Provider
+            {...provider}
+            UNSAFE_className={classNames(listStyles, 'react-spectrum-ListViewItem', 'is-dragging')}
+            UNSAFE_style={{width: itemWidth, paddingInlineStart: 0}}>
+            <div className={listStyles['react-spectrum-ListViewItem-grid']} data-testid="dragpreview">
+              <div className={listStyles['react-spectrum-ListViewItem-draghandle-container']}>
+                <div className={listStyles['react-spectrum-ListViewItem-draghandle-button']}>
+                  <DragHandle />
+                </div>
+              </div>
+              {showCheckbox &&
+                <Checkbox
+                  isSelected={isSelected}
+                  UNSAFE_className={listStyles['react-spectrum-ListViewItem-checkbox']}
+                  isEmphasized
+                  aria-label={checkboxProps['aria-label']} />
+              }
+              <SlotProvider
+                slots={{
+                  content: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-content']},
+                  text: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-content']},
+                  description: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-description']},
+                  icon: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-icon'], size: 'M'},
+                  image: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-image']},
+                  link: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-content'], isQuiet: true},
+                  actionButton: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-actions'], isQuiet: true},
+                  actionGroup: {
+                    UNSAFE_className: listStyles['react-spectrum-ListViewItem-actions'],
+                    isQuiet: true,
+                    density: 'compact'
+                  },
+                  actionMenu: {UNSAFE_className: listStyles['react-spectrum-ListViewItem-actionmenu'], isQuiet: true}
+                }}>
+                {typeof item.rendered === 'string' ? <Content>{item.rendered}</Content> : item.rendered}
+              </SlotProvider>
+            </div>
+          </Provider>
+        );
+      },
+      itemAllowsDragging: () => true
+    });
+  }
+
+  let dragHooksProvided = useRef(isListDraggable);
+  useEffect(() => {
+    if (!dragHooksProvided.current && isListDraggable) {
+      console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
+    }
+  }, [isListDraggable]);
```

Comment:
It looks like React throws an error before we can give a warning in this case. Might need to find a better way to do this.

Reply:
Perhaps move this check up below the `isListDraggable` check and ditch the useEffect, just have it check during each render. 

maybe something like:
```
  let isListDraggable = !!dragHooks;
  let dragHooksProvided = useRef(isListDraggable);
  if (dragHooksProvided.current && !isListDraggable) {
    console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
  }
```